src/overlays/actors/ovl_Bg_F40_Switch/z_bg_f40_switch.c

```diff
@@ -34,37 +35,160 @@ const ActorInit Bg_F40_Switch_InitVars = {
     (ActorFunc)BgF40Switch_Draw,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80BC4E04[] = {
+s32 sBgF40SwitchGlobalsInitialized = false;
+u32 sBgF40SwitchLastUpdateFrame;
+
+extern CollisionHeader D_06000118;
+extern Gfx D_06000438;
+extern Gfx D_06000390;
+
+// Updates all instances of this actor in the current room, unless it's already been called this frame
+void func_80BC47B0(BgF40Switch* this, GlobalContext* globalCtx) {
+    if (globalCtx->gameplayFrames != sBgF40SwitchLastUpdateFrame) {
+        u32 pressedSwitchFlags[4] = { 0 };
+        u32 pad;
+        s32 switchFlagIndex;
+        s32 isPressedNow;
+        Actor* actor;
+        BgF40Switch* actorAsSwitch;
+        s32 unk40 = func_801233E4(globalCtx);
+        for (actor = globalCtx->actorCtx.actorList[0].first; actor; actor = actor->next) {
+            if (actor->id == ACTOR_BG_F40_SWITCH && actor->room == this->actor.actor.room && actor->update != 0) {
+                actorAsSwitch = (BgF40Switch*)actor;
+                actorAsSwitch->wasPressed = actorAsSwitch->isPressed;
+                isPressedNow = DynaPolyActor_IsInSwitchPressedState(&actorAsSwitch->actor);
+                if (actorAsSwitch->isPressed && actorAsSwitch->actionFunc == BgF40Switch_ActionIdlePressed) {
+                    // Switch is fully pressed - if there's nothing keeping it pressed, wait a short time before
+                    // reverting to unpressed state.
+                    if (isPressedNow || unk40) {
+                        actorAsSwitch->switchReleaseDelay = 6;
+                    } else {
+                        if (actorAsSwitch->switchReleaseDelay > 0) {
+                            --actorAsSwitch->switchReleaseDelay;
+                        } else {
+                            actorAsSwitch->isPressed = isPressedNow;
+                        }
+                    }
+                } else {
+                    // No delay when pressing switch, or releasing from a not-fully-pressed state.
+                    actorAsSwitch->isPressed = isPressedNow;
+                }
+                if (actorAsSwitch->isPressed) {
+                    switchFlagIndex = (actorAsSwitch->actor.actor.params & 0xFE00) >> 9;
+                    if (switchFlagIndex >= 0 && switchFlagIndex < 0x80) {
+                        pressedSwitchFlags[(switchFlagIndex & ~0x1F) >> 5] |= 1 << (switchFlagIndex & 0x1F);
```

Comment:
Here too. Any access to params as a pseudo-bitfield needs a macro.

Reply:
In z_actor.c these are used without macros. Do I need to update the whole codebase?